Source/Projections/Builder/UnregisteredProjections.cs

```diff
@@ -68,4 +75,17 @@ public UnregisteredProjections(IUniqueBindings<ProjectionModelId, IProjection> p
             projectionConverter,
             loggerFactory.CreateLogger(processorType)) as EventProcessor<ProjectionId, ProjectionRegistrationRequest, ProjectionRequest, ProjectionResponse>;
     }
+    void AddToContainer(TenantId tenantId, IServiceCollection serviceCollection)
+    {
+        foreach (var projection in Values)
+        {
+            var readModelType = projection.ProjectionType;
+            serviceCollection.AddSingleton(
+                typeof(IProjectionOf<>).MakeGenericType(readModelType),
+                serviceProvider => Activator.CreateInstance(
+                    typeof(ProjectionOf<>).MakeGenericType(readModelType),
+                    serviceProvider.GetService<IProjectionStore>(),
+                    new ScopedProjectionId(projection.Identifier, projection.ScopeId)) ?? throw new CouldNotCreateProjectionOf(projection.ProjectionType, tenantId));
```

Comment:
In my mind it would be simpler here to just get the IProjectionStore from the service provider and call `.Of(readModelType)` ? Then we need less reflection magic, and maybe not the new exception?

Reply:
This is resolved